### DISCUSSION TITLE
How to use array inputs parameters inside expression?

### DISCUSSION THREAD

**Discussion by pavan02** [argoproj/argo-workflows#13925] [Q&A]
but works with hard coded array
`when: {{=item.dir in ["dir1","dir2","dir3"]}}`

doesn't work when input parameters dirs is `["dir1", "dir2", "dir3"]`  .. errors out with `failed to evaluate expression: unknown name inputs (1:13) | item.dir in inputs.parameters.dirs | ............^`
`when: {{=item.dir in inputs.parameters.dirs}}`

**Comment by jswxstw** [comment]
You should transform `inputs.parameters.dirs` from `String`  to `Array`, like `when: {{=item.dir in fromJSON(inputs.parameters.dirs)}}`.

**Reply by pavan02** [reply]
still the same error 😢 

```
failed to evaluate expression: unknown name inputs (1:22) | item.dir in fromJSON(inputs.parameters.dirs)
```

**Reply by jswxstw** [reply]
Could you provide the complete workflow?
